receiver/otlpreceiver/otlp.go

```diff
@@ -50,6 +54,39 @@ type otlpReceiver struct {
 	startServerOnce sync.Once
 }
 
+// httpBodyCompressionHandler is a HTTP middleware checks the "Content-Encoding" HTTP header and if
+// a compression such as "gzip", "deflate", "zlib", is found, the request body will
+// be uncompressed accordingly. It it encounters an error while do is, the original request is passed untouched.
+func httpBodyCompressionHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(r.Body)
+			if err != nil {
+				break
+			}
+			defer gr.Close()
+			body, err := ioutil.ReadAll(gr)
+			if err != nil {
+				break
+			}
+			r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
Is it valid to replace the `Body` of the request? Does documentation say anything about it?

From https://pkg.go.dev/net/http?tab=doc#Handler
>Except for reading the body, handlers should not modify the provided Request.

Reply:
@tigrannajaryan Oh that's not good. I had a feeling this is a hacky solution. If this approach won't work, I think the only alternative is to have this implemented as a feature in `grpc-gateway`, similar to how `Marshaller` that works based on the `Content-Type` header.